### PR TITLE
Implementation of responsive DOM

### PR DESCRIPTION
**Pull request by parthlambdatest**

### CODE REVIEW

File: src/lib/processSnapshot.ts [hunk 1]
```diff
@@ -16,28 +16,294 @@ export default class Queue {
...
+            for (const device of devices) {
+                const variant = `${snapshot.name}_${device}_${orientation}_${fullPage ? 'fullPage' : 'noFullPage'}`;
+                this.variants.push(variant);
+            }
+        }
+    }
+    
+
+    private generateWebVariants(snapshot: Snapshot, webConfig: any): void {
+        const browsers = webConfig.browsers ?? this.ctx.config.web?.browsers ?? ["chrome", "edge", "firefox", "safari"];
```
Comment (sushobhit-lt) [inline src/lib/processSnapshot.ts:91]: @parthlambdatest Please use static constants for browsers

File: src/tasks/processSnapshot.ts [hunk 2]
```diff
@@ -10,6 +10,10 @@ export default (ctx: Context): ListrTask<Context, ListrRendererFactory, ListrRen
 
             try {
                 // wait for snapshot queue to be empty
+                if(ctx.config.deferUploads){
+                    console.log("started after processing because of deferuploads")
```
Comment (sushobhit-lt) [inline src/tasks/processSnapshot.ts:14]: Use ctx.log.info(""), here

File: src/lib/processSnapshot.ts [hunk 3]
```diff
@@ -16,28 +16,294 @@ export default class Queue {
...
+                    const variant = `${snapshot.name}_${browser}_viewport[${width}]_viewport[${height}]`;
+                    this.variants.push(variant);
+                }
+            }
+        }
+    
+        // Process mobile configurations if they exist
+        if (config.mobile) {
+            const devices = config.mobile.devices || [];
+            const orientation = config.mobile.orientation || "portrait";  // Default to portrait if not provided
```
Comment (sushobhit-lt) [inline src/lib/processSnapshot.ts:79]: Use  `"portrait"` as a constant , its used 6 time in this file

File: src/lib/processSnapshot.ts [hunk 4]
```diff
@@ -16,28 +16,294 @@ export default class Queue {
...
+            const fullPage = config.mobile.fullPage ?? true; // FullPage defaults to true if not defined
+            
+            for (const device of devices) {
+                const variant = `${snapshot.name}_${device}_${orientation}_${fullPage ? 'fullPage' : 'noFullPage'}`;
```
Comment (sushobhit-lt) [inline src/lib/processSnapshot.ts:83]: Please use static constants for browsers `'fullPage'`  `'noFullPage'` values, its used 4 times in same file

**Review by sushobhit-lt — APPROVED** [review]

File: src/lib/processSnapshot.ts [hunk 5]
```diff
@@ -16,28 +16,290 @@ export default class Queue {
...
     private ctx: Context;
+    private snapshotNames: Array<string> = [];
+    private variants: Array<string> = [];
 
     constructor(ctx: Context) {
         this.ctx = ctx;
     }
 
-    enqueue(item: Snapshot): void {
+    enqueue(item: Snapshot, start: boolean): void {
```
Comment (pinanks) [inline src/lib/processSnapshot.ts:26]: This is redundant. Use ctx.config which is already in the queue.

**Review by pinanks — CHANGES_REQUESTED** [review]

**Comment by pinanks** [comment]
Also, add the Queue in a separate file called `snapshotQueue.ts`

**Review by pinanks — APPROVED** [review]

**Pull request merged**